client/state/ui/editor/selectors.js

```diff
@@ -27,3 +38,35 @@ export function isEditorNewPost( state ) {
 export function isEditorDraftsVisible( state ) {
 	return state.ui.editor.showDrafts;
 }
+
+/**
+ * Returns the editor URL path for the given site ID, post ID pair.
+ *
+ * @param  {Object} state  Global state tree
+ * @param  {Number} siteId Site ID
+ * @param  {Number} postId Post ID
+ * @return {String}        Editor URL path
+ */
+export function getEditorPath( state, siteId, postId ) {
+	const type = get( getSitePost( state, siteId, postId ), 'type', 'post' );
+
+	let path;
+	switch ( type ) {
+		case 'post': path = '/post'; break;
+		case 'page': path = '/page'; break;
+		default: path = `/type/${ type }`; break;
```

Comment:
>I think `/edit/${ type }` is a better choice here.

Ignoring everything else, I'd agree that `/edit/${ type }` is a nice path. It does lose the consistency we have with `/posts` -> `/post` and `/pages` -> `/page`. I'm still not totally content with the `/types` path either at this point (#3689).

>Perhaps also `/view/${ type }` rather than `/types/${ type }` for the listing page?

"View" seems like a pretty loaded term that'd be ripe for conflict, and it's not immediately obvious it relates to posts (vs. something else, customizer, theme preview, etc). In any case, `/list` would probably be more appropriate (since it's not viewing a single item), but suffers from similar concerns.

For reference, the wp-admin paths are:
- List view: `/edit.php?post_type=jetpack-testimonial`
- Detail view: `/edit.php?post=1&action=edit`

I don't particularly like these paths, but worth highlighting nonetheless.

We're still early enough in the process that we can iterate on the path, especially as more people come to find the page and have their own opinions.